### PR TITLE
QPPSF-2996 EMA updates 2018

### PR DESCRIPTION
**Pull request by rockykitamura**
https://jira.cms.gov/browse/QPPSF-2996

Updates to the EMA clinical clusters and specialty sets for 2018
- Modified `getClinicalClusterData` and `getClinicalClusterSchema` to accept performanceYear parameter, defaults to 2017
- Updates to the EMA cluster information based on https://confluence.cms.gov/pages/viewpage.action?spaceKey=QPPPRACTICE&title=2018+EMA+Logic+and+Set

SPECIALTY SETS FOR 2018:
```
{
    "claims": [{
        "name": "allergyImmunology",
        "measureIds": ["110", "111", "130", "226", "317"]
    }, {
        "name": "anesthesiology",
        "measureIds": ["076", "130", "317"]
    }, {
        "name": "dermatology",
        "measureIds": ["130", "226", "317"]
    }, {
        "name": "generalOncology",
        "measureIds": ["047", "130", "226", "250", "317"]
    }, {
        "name": "hospitalists",
        "measureIds": ["047", "076", "130", "407"]
    }, {
        "name": "interventionalRadiology",
        "measureIds": ["076", "145", "437"]
    }, {
        "name": "neurosurgical",
        "measureIds": ["021", "023", "130", "226"]
    }, {
        "name": "pediatrics",
        "measureIds": ["091", "093", "110", "134"]
    }, {
        "name": "plasticSurgery",
        "measureIds": ["021", "023", "130", "226", "317"]
    }, {
        "name": "podiatry",
        "measureIds": ["128", "154", "155", "226"]
    }, {
        "name": "radiationOncology",
        "measureIds": ["156"]
    }, {
        "name": "rheumatology",
        "measureIds": ["047", "128", "130", "226", "317"]
    }],
    "registry": [{
        "name": "electrophysiologyCardiacSpecialist",
        "measureIds": ["348", "392", "393"]
    }, {
        "name": "radiationOncology",
        "measureIds": ["102", "143", "144", "156"]
    }]
}
```

CLINICAL CLUSTERS FOR 2018:
```
{
    "claims": [{
        "name": "appropriateFollowUpImaging",
        "measureIds": ["405", "406"]
    }, {
        "name": "chiropracticCare",
        "measureIds": ["131", "182"]
    }, {
        "name": "colonoscopyCare",
        "measureIds": ["130", "425"]
    }, {
        "name": "diabeticCare",
        "measureIds": ["001", "117", "128"]
    }, {
        "name": "earNoseThroatCare",
        "measureIds": ["091", "093"]
    }, {
        "name": "emergencyCare",
        "measureIds": ["254", "255"]
    }, {
        "name": "endoscopyAndPolypSurveillance",
        "measureIds": ["185", "320"]
    }, {
        "name": "generalCare",
        "measureIds": ["130", "226", "134", "317"]
    }, {
        "name": "headacheCare",
        "measureIds": ["419", "435"]
    }, {
        "name": "immunizationCare",
        "measureIds": ["110", "111"]
    }, {
        "name": "lungCare",
        "measureIds": ["051", "052", "130"]
    }, {
        "name": "pathology",
        "measureIds": ["099", "100", "249", "250", "251"]
    }, {
        "name": "pathologyLungCancer",
        "measureIds": ["395", "396"]
    }, {
        "name": "preventiveCare",
        "measureIds": ["130", "317", "112"]
    }, {
        "name": "preventiveCare",
        "measureIds": ["130", "317", "113"]
    }, {
        "name": "surgicalCare",
        "measureIds": ["021", "023", "047"]
    }, {
        "name": "urinaryIncontinenceCare",
        "measureIds": ["048", "050"]
    }],
    "registry": [{
        "name": "acuteOtitisExterna",
        "measureIds": ["091", "093"]
    }, {
        "name": "adultRenalCatheterCare",
        "measureIds": ["329", "330"]
    }, {
        "name": "adultRenalDiseaseCare",
        "measureIds": ["110", "122", "130", "226"]
    }, {
        "name": "adultSinusitisCare",
        "measureIds": ["331", "332", "333", "334"]
    }, {
        "name": "ageRelatedMacularDegeneration",
        "measureIds": ["014", "140"]
    }, {
        "name": "anesthesiologyCare",
        "measureIds": ["424", "430", "426"]
    }, {
        "name": "anesthesiologyCare",
        "measureIds": ["424", "430", "427"]
    }, {
        "name": "appropriateTestTreatmentForChildren",
        "measureIds": ["065", "066"]
    }, {
        "name": "asthmaCare",
        "measureIds": ["110", "130", "226", "398"]
    }, {
        "name": "asthmaCare",
        "measureIds": ["110", "130", "226", "444"]
    }, {
        "name": "breastSurgeryCare",
        "measureIds": ["262", "263", "264"]
    }, {
        "name": "cancerEndOfLife",
        "measureIds": ["453", "454", "455", "456"]
    }, {
        "name": "cardiacStressImaging",
        "measureIds": ["322", "323", "324"]
    }, {
        "name": "chronicObstructivePulmonaryDiseaseCopdCare",
        "measureIds": ["130", "051"]
    }, {
        "name": "chronicObstructivePulmonaryDiseaseCopdCare",
        "measureIds": ["130", "052"]
    }, {
        "name": "colonoscopyCare",
        "measureIds": ["130", "425"]
    }, {
        "name": "coronaryArteryBypassGraftCare",
        "measureIds": ["165", "166", "167", "168", "445"]
    }, {
        "name": "coronaryDiseaseCare",
        "measureIds": ["130", "226", "006"]
    }, {
        "name": "coronaryDiseaseCare",
        "measureIds": ["130", "226", "007"]
    }, {
        "name": "coronaryDiseaseCare",
        "measureIds": ["130", "226", "118"]
    }, {
        "name": "dementia",
        "measureIds": ["282", "283", "286", "288"]
    }, {
        "name": "diabetesMellitusFootCare",
        "measureIds": ["126", "127"]
    }, {
        "name": "endoscopyAndPolypSurveillance",
        "measureIds": ["185", "320", "439"]
    }, {
        "name": "gynecologicalCare",
        "measureIds": ["422", "432", "433", "434"]
    }, {
        "name": "headacheCare",
        "measureIds": ["419", "435"]
    }, {
        "name": "heartFailureCare",
        "measureIds": ["130", "005"]
    }, {
        "name": "heartFailureCare",
        "measureIds": ["130", "008"]
    }, {
        "name": "heartRhythmCare",
        "measureIds": ["348", "392", "393"]
    }, {
        "name": "hematologyCare",
        "measureIds": ["067", "068", "069", "070"]
    }, {
        "name": "immunizationCare",
        "measureIds": ["110", "111"]
    }, {
        "name": "inflammatoryBowelDisease",
        "measureIds": ["271", "275"]
    }, {
        "name": "melanomaCare",
        "measureIds": ["137", "138", "224"]
    }, {
        "name": "opioidCare",
        "measureIds": ["408", "412", "414"]
    }, {
        "name": "optimizingPatientExposureToIonizingRadiation",
        "measureIds": ["359", "360", "361"]
    }, {
        "name": "osteoporosisCare",
        "measureIds": ["130", "226", "024"]
    }, {
        "name": "osteoporosisCare",
        "measureIds": ["130", "226", "418"]
    }, {
        "name": "palliativeCare",
        "measureIds": ["047", "134", "342"]
    }, {
        "name": "parkinsonsDisease",
        "measureIds": ["290", "291", "293"]
    }, {
        "name": "pathologyBreastCancer",
        "measureIds": ["099", "251"]
    }, {
        "name": "pathologyLungCancer",
        "measureIds": ["395", "396"]
    }, {
        "name": "pediatricKidneyDiseaseCare",
        "measureIds": ["327", "328"]
    }, {
        "name": "perioperativeCare",
        "measureIds": ["021", "023"]
    }, {
        "name": "preventiveCare",
        "measureIds": ["130", "317", "112"]
    }, {
        "name": "preventiveCare",
        "measureIds": ["130", "317", "113"]
    }, {
        "name": "preventiveCare",
        "measureIds": ["130", "317"]
    }, {
        "name": "primaryOpenAngleGlaucoma",
        "measureIds": ["012", "141"]
    }, {
        "name": "retinalCare",
        "measureIds": ["384", "385"]
    }, {
        "name": "totalKneeReplacement",
        "measureIds": ["350", "351", "352", "353"]
    }, {
        "name": "urinaryIncontinenceCare",
        "measureIds": ["048", "050"]
    }]
}
```

### CODE REVIEW

File: scripts/clinical-clusters/build-clinical-clusters [hunk 1]
```diff
@@ -1,11 +1,12 @@
 #!/usr/bin/env bash
 
+currentPerformanceYear=2018
```
Comment (kalvinwang) [inline scripts/clinical-clusters/build-clinical-clusters]: No problem if this is what makes sense here, but I think in other files we pass the perfYear as an argument to the build script-- is it better to just set it directly here?

File: scripts/clinical-clusters/ema-clinical-cluster-builder.js [hunk 2]
```diff
@@ -25,53 +25,129 @@ const MAX_SPECIALITY_SET_SIZE = 6;
...
+  // handle removal of sets/measures
+  if (relations) {
+    relations.forEach(r => {
+      if (r.action === 'remove') {
+        r.measureIds.forEach(m => {
+          const specialtySets = clusterMap.get(m).specialtySets;
+          const specialtySetIndex = specialtySets.findIndex(ss => ss.name === r.name);
+          specialtySets.splice(specialtySetIndex, 1);
+        });
+      } else if (r.action === 'override') {
```
Comment (kalvinwang) [inline scripts/clinical-clusters/ema-clinical-cluster-builder.js]: If there are any useful comments for the intent of this function that'd be great to add, particularly for "override" (as a reader with no context on any of this clinical clusters code)

Comment (rockykitamura) [reply scripts/clinical-clusters/ema-clinical-cluster-builder.js]: Thanks for the feedback!  I'll add some comments to make my intent more obvious.

File: scripts/clinical-clusters/build-clinical-clusters (hunk 1 above)
Comment (kalvinwang) [reply scripts/clinical-clusters/build-clinical-clusters]: Gotcha, thank you!

**Pull request merged**